### PR TITLE
feat(oui-header-tabs): add on click events

### PR DESCRIPTION
**Pull request by varun257**
### Description of the Change

add on-click event for header-tabs-item
add on-next-slide event oui-slideshow

### Benefits

More options

### Possible Drawbacks

None

### Applicable Issues

Helps resolve MBE-283, MBE-284

### CODE REVIEW

File: packages/oui-header-tabs/README.md [hunk 1]
```diff
@@ -97,6 +97,7 @@
 | `active`      | boolean   | <?        | no                | `true`, `false`   | `false`   | manual active flag
 | `disabled`    | boolean   | <?        | yes               | `true`, `false`   | `false`   | disabled flag
 | `external`    | boolean   | <?        | yes               | `true`, `false`   | `false`   | external link flag
+| `on-click`    | function  | &         | no                | n/a               | n/a       | handler triggered when clicked
```
Comment (FredericEspiau) [inline packages/oui-header-tabs/README.md]: Please use it in an example ;)

File: packages/oui-slideshow/src/slideshow.component.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ export default {
         onDismiss: "&",
         loading: "<?",
         loop: "<?",
-        theme: "@?"
+        theme: "@?",
+        onNext: "&"
```
Comment (FredericEspiau) [inline packages/oui-slideshow/src/slideshow.component.js]: Please use it in an example ;)

Comment (AxelPeter) [reply packages/oui-slideshow/src/slideshow.component.js]: Yes, with a note of what is returned with this event (eg. `index`)

File: packages/oui-slideshow/README.md [hunk 3]
```diff
@@ -88,6 +88,7 @@
 | `loading`     | boolean   | <?        | no                | `true`, `false`   | `false`   | display loader flag
 | `loop`        | boolean   | <?        | no                | `true`, `false`   | `false`   | whether the component should cycle continuously
 | `on-dismiss`  | function  | &         | no                | n/a               | n/a       | dismiss callback
+| `on-next`     | function  | &         | no                | n/a               | n/a       | handler triggered when on click of next slide
```
[the remark this answers is not in the document]
Comment (FredericEspiau) [reply packages/oui-slideshow/README.md]: `to next` is a verb for your information, but it doesn't mean what you think 😄

**Pull request closed without merging**